apps/src/lib/levelbuilder/lesson-editor/FindProgrammingExpressionDialog.jsx

```diff
@@ -181,9 +168,18 @@ class FindProgrammingExpressionDialog extends Component {
 
     fetch('/programmingexpressionsearch?' + new URLSearchParams(params))
       .then(response => response.json())
-      .then(data => this.setState({programmingExpressions: data}));
+      .then(data => {
+        this.setState({
+          programmingExpressions: data.programmingExpressions,
+          numPages: data.numPages
+        });
+      });
   }
 
+  setCurrentPage = value => {
+    this.setState({currentPage: value}, this.doSearch);
```

Comment:
Instead of passing `doSearch` as a callback, we should add `currentPage` to the set of parameters checked by `componentDidUpdate`